### DISCUSSION TITLE
Does navidrome support Lyrics?

### DISCUSSION THREAD

**Discussion by zslzsl999** [navidrome/navidrome#2603] [General]
I just installed navidrome server on my nas and it's so convenient.
But I have noticed that there is "toggle lyrics" button and it shows "no lyrics" on the website after I clicked.
Does navidrome support Lyrics?  both on website and apps?
Should I download the lrc file and set the config, or navidrome will search lyrics automaticly?

**Comment by dweymouth** [comment]
Navidrome currently only supports lyrics embedded as tags in the file. And only synced lyrics show in the Web UI though either synced or unsynced lyrics can be shown in other clients.

There is an open feature request for supporting .lrc files. Navidrome being able to automatically fetch lyrics from 3rd party sources would be really nice too but not sure it's planned at the moment.

**Comment by GetOutOfMyBakery** [comment]
On the off-chance someone else lands here (like I did when searching for the same topic), this has since been introduced:
- https://github.com/navidrome/navidrome/pull/2897

Released as part of [v0.56.0](https://github.com/navidrome/navidrome/releases/tag/v0.56.0)

Tagging: @zslzsl999 @dweymouth @PeterCodar @jvearner

As an added bonus, here's [some guidance](https://github.com/tranxuanthang/lrcget/issues/207#issuecomment-3383159309) around getting `lrcget` running in a docker container to handle downloading `.lrc` files 👍

> [!NOTE]
> The `.lrc` file support is only on Subsonic clients (ala Symphonium, etc.) not on the Navidrome WebUI.

For more context, see:
- [[BUG]: External .lrc file not showing up on Navidrome web](https://github.com/navidrome/navidrome/issues/4148)

> This is not a "bug", it is a limitation of the current player, and won't be fixed in the current UI. External .lrc lyrics are only available in Subsonic clients that supports it, ex: Feishin and Supersonic. 
https://github.com/navidrome/navidrome/issues/4148#issuecomment-2936319726

**Reply by iMonZ** [reply]
> On the off-chance someone else lands here (like I did when searching for the same topic), this has since been introduced:
> 
> * [[enhancement]: Support fetching lyrics from filesystem #2897](https://github.com/navidrome/navidrome/pull/2897)
> 
> Released as part of [v0.56.0](https://github.com/navidrome/navidrome/releases/tag/v0.56.0)
> 
> Tagging: @zslzsl999 @dweymouth @PeterCodar @jvearner
> 
> As an added bonus, here's [some guidance](https://github.com/tranxuanthang/lrcget/issues/207#issuecomment-3383159309) around getting `lrcget` running in a docker container to handle downloading `.lrc` files 👍
> 
> Note
> 
> The `.lrc` file support is only on Subsonic clients (ala Symphonium, etc.) not on the Navidrome WebUI.
> 
> For more context, see:
> 
> * [[BUG]: External .lrc file not showing up on Navidrome web](https://github.com/navidrome/navidrome/issues/4148)
> 
> > This is not a "bug", it is a limitation of the current player, and won't be fixed in the current UI. External .lrc lyrics are only available in Subsonic clients that supports it, ex: Feishin and Supersonic.
> > [#4148 (comment)](https://github.com/navidrome/navidrome/issues/4148#issuecomment-2936319726)

Thanks! 
Is there any plan or open issue to support for automatic lyrics download?

**Reply by deluan** [reply]
Yes, it is planned, but probably as a plugin.

**Reply by dot-file** [reply]
> Yes, it is planned, but probably as a plugin.

Any updates on this?

**Reply by Pizzakira** [reply]
Any other updates ?

**Reply by deluan** [reply]
https://github.com/J0R6IT0/navidrome-lrclib-plugin

**Reply by ElEmFurt** [reply]
> https://github.com/J0R6IT0/navidrome-lrclib-plugin

How do you use this though? I mean I set it up after going through the instructions on this plugin's page and adding the lyrics variable to the compose as mentioned and the config of the plugin in Navidrome, but clicking the lyrics button in the navidrome web interface doesnt fetch the lyrics.

**Reply by deluan** [reply]
The current web UI is very limited in terms of lyrics. The plugin works fine for any clients that shows lyrics, but not the web UI.
